src/main/java/org/craftercms/studio/impl/v1/service/site/SiteServiceImpl.java

```diff
@@ -1782,6 +1782,20 @@ private void addSiteUuidFile(String site, String siteUuid) throws IOException {
         Files.write(path, toWrite.getBytes());
     }
 
+    @Override
+    public boolean lockPublishingForSite(String siteId, String lockOwnerId) {
+        logger.debug("Locking publishing for site " + siteId + " with lock owner " + lockOwnerId);
```

Comment:
Replace with a stored procedure that does everything in an atomic operation (test and set)